File: src/dataset/cpp/isogram_Exercism_thefullarcticfox/claude/ClaudeSonnet4_isogram_v1.cpp
#include "isogram.h"
#include <unordered_set>
#include <cctype>

namespace isogram {
	bool	is_isogram(const std::string& s) {
		if (s.empty())
			return true;

		std::unordered_set<char> seen;
		seen.reserve(26); // Maximum 26 letters in alphabet
		
		for (char c : s) {
			if (c != ' ' && c != '-') {
				char lower_c = std::tolower(static_cast<unsigned char>(c));
				if (!seen.insert(lower_c).second) {
					return false;
				}
			}
		}
		return true;
	}
}  // namespace isogram

File: src/dataset/typescript/isogram_exercism-javascript-samajammin/claude/ClaudeSonnet4_isogram_v1.ts
export default class Isogram {
  public static isIsogram(input: string): boolean {
    const seen = new Set<string>();
    
    for (let i = 0; i < input.length; i++) {
      const char = input[i];
      if (char !== '-' && char !== ' ') {
        const normalized = char.toLowerCase();
        if (seen.has(normalized)) {
          return false;
        }
        seen.add(normalized);
      }
    }
    
    return true;
  }
}